Dockerfile

```diff
@@ -1,32 +1,30 @@
+# syntax = docker/dockerfile:1.2
 FROM golang:1.16.2-buster
 
+# install dependencies.
 RUN apt update && \
     apt-get install -y \
         build-essential \
         ca-certificates \
-        unzip \
         curl
-
-# INSTALL NODE
 RUN curl -sL https://deb.nodesource.com/setup_14.x | bash - && \
     apt-get install -y nodejs
 
-# COPY STARPORT SOURCE CODE INTO CONTAINER
-COPY ./docs /starport/docs
-COPY ./starport /starport/starport
-COPY ./go.mod /starport/go.mod
-COPY ./go.sum /starport/go.sum
+# enable faster module downloading.
+ENV GOPROXY https://proxy.golang.org
+
 WORKDIR /starport
 
-# INSTALL STARPORT
-RUN go install -mod=readonly ./...
+# cache dependencies.
+COPY ./go.mod . 
+COPY ./go.sum . 
+RUN go mod download
 
-# ENTRYPOINT
-ENTRYPOINT ["/go/bin/starport"]
+# build Starport.
+COPY . .
+# WARNING!
+# use `DOCKER_BUILDKIT=1` with `docker build` to enable --mount feature.
+RUN --mount=type=cache,target=/root/.cache/go-build go install -v ./...
+RUN rm -rf /starport
 
-# EXPOSE PORTS
-EXPOSE 12345
```

Comment:
I would keep them to make explicit to users those are the reachable ports

Reply:
added back e1432f0.